lib/api/objectHead.js

```diff
@@ -77,8 +78,47 @@ function objectHead(authInfo, request, log, callback) {
             if (headerValResult.error) {
                 return callback(headerValResult.error, corsHeaders);
             }
+            const responseHeaders = collectResponseHeaders(objMD, corsHeaders,
```

Comment:
This method is quite long now ( roughly 80 lines ) and although eslint doesn't require it I would suggest splitting it up if possible. I generally recommend no more than about a "screen" tall. In practice this is varied but 20-40 lines is a good range in most cases for me but would, of course, adjust based on language or needs.

Reply:
I agree about this; for now, I copied the logic from the objectGet API as it is, in order to to keep those consistent and not to impact the scope of this PR.
I plan to create a follow-up ticket to submit a refactor PR to address your comment as well as some other closely related issues that I encountered.